File: src/app/src/main/java/jp/co/e2/dogage/adapter/PetAgeFragmentPagerAdapter.java
package jp.co.e2.dogage.adapter;

import java.util.ArrayList;

import jp.co.e2.dogage.activity.PetAgeFragment;
import jp.co.e2.dogage.entity.PetEntity;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

/**
 * ページアダプタ―
 */
public class PetAgeFragmentPagerAdapter extends FragmentStatePagerAdapter {
    private ArrayList<PetEntity> mList;

    /**
     * コンストラクタ
     *
     * @param fm FragmentManager
     */
    public PetAgeFragmentPagerAdapter(FragmentManager fm, ArrayList<PetEntity> list) {
        super(fm);

        mList = list;
    }

    /**
     * ${inheritDoc}
     */
    @Override
    public Fragment getItem(int position) {
        return PetAgeFragment.newInstance(position, mList.get(position));
    }

    /**
     * ${inheritDoc}
     */
    @Override
    public int getCount() {
        return mList.size();
    }
}

File: src/app/src/main/java/jp/co/e2/dogage/adapter/PetAgeFragmentPagerAdapter.kt
package jp.co.e2.dogage.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import java.util.ArrayList

import jp.co.e2.dogage.activity.PetAgeFragment
import jp.co.e2.dogage.entity.PetEntity

/**
 * ページアダプタ―
 */
class PetAgeFragmentPagerAdapter(
        fm: FragmentManager,
        private val list: ArrayList<PetEntity>
) : FragmentStatePagerAdapter(fm) {

    /**
     * ${inheritDoc}
     */
    override fun getItem(position: Int): Fragment {
        return PetAgeFragment.newInstance(position, list[position])
    }

    /**
     * ${inheritDoc}
     */
    override fun getCount(): Int {
        return list.size
    }
}